src/main/java/com/amazonaws/encryptionsdk/kms/KmsUtils.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except
+ * in compliance with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.amazonaws.encryptionsdk.kms;
+
+import com.amazonaws.arn.Arn;
+import com.amazonaws.encryptionsdk.exception.MalformedArnException;
+import com.amazonaws.services.kms.AWSKMS;
+
+public class KmsUtils {
+
+    private static final String ALIAS_PREFIX = "alias/";
+    /**
+     * The provider ID used for the KmsKeyring
+     */
+    public static final String KMS_PROVIDER_ID = "aws-kms";
+
+    /**
+     * Parses region from the given arn (if possible) and passes that region to the
+     * given clientSupplier to produce an {@code AWSKMS} client.
+     *
+     * @param arn            The Amazon Resource Name or Key Alias
+     * @param clientSupplier The client supplier
+     * @return AWSKMS The client
+     * @throws MalformedArnException if the arn is malformed
+     */
+    public static AWSKMS getClientByArn(String arn, KmsClientSupplier clientSupplier) throws MalformedArnException {
```

Comment:
@seebees I believe this behavior is the same as the one proposed in https://github.com/aws/aws-encryption-sdk-javascript/pull/231, however I suggest that you take a close look here to ensure that the KMS Keyring clientSupplier logic is consistent (as well as with https://github.com/awslabs/aws-encryption-sdk-specification/pull/50)

Reply:
This only somewhat follows the spec.
The spec says that if a region can not be extracted, then we should pass an unknown identifier.
In the case of a `key/01234567-89ab-cdef-fedc-ba9876543210` or `01234567-89ab-cdef-fedc-ba9876543210` a region can not be extracted...

Reply:
The latest changes ensure raw Key IDs are supported for KmsMasterKey. KmsKeyring still does not support raw key IDs, that is pending a decision on whether we want to support them for key rings